### ISSUE TITLE
Repeated repeatable directives on extensions are rejected

### ISSUE THREAD

**Issue by robertpanzer** [graphql-java/graphql-java#2146]
**Describe the bug**
graphql-java correctly accepts schemas with repeated repeatable directives on a type.
I.e. this schema is accepted:
```graphql
directive @key(fields:String!) repeatable on OBJECT
type Query{  whoami: String!  parameterized(x: String!): PType }
type PType @key(fields:"whoami") @key(fields:"whoami") {   name: String }
```
However, this schema is rejected:
```graphql
directive @key(fields:String!) repeatable on OBJECT
type Query{  whoami: String!  parameterized(x: String!): PType }
type PType @key(fields:"whoami") @key(fields:"whoami") {   name: String }
extend type PType @key(fields:"whoami") { somethingelse: String }
```
with this error:
```
SchemaProblem{errors=[The extension 'PType' type [@3:1] has redefined the directive called 'key', 'PType' extension type [@4:1] tried to redefine field 'name' [@4:45]]
}
```

Imo the latest spec draft allows this though: https://spec.graphql.org/draft/#sel-IAHZnCRFHABQB5CnlY

> Any non‐repeatable directives provided must not already apply to the original Object type.

This changed to the 2018 version which stated this:

> Any directives provided must not already apply to the original Object type.

**To Reproduce**
This simple example shows the problem, I used a simple MockedWiringFactory that I found in some other issue here.

```java
public class QueryValidatorTest {

    private static final String schema =
            "directive @key(fields:String!) repeatable on OBJECT\n"
            + "type Query{\n"
            + " whoami: String!\n"
            + " parameterized(x: String!): PType\n"
            + "}\n"
            + "type PType @key(fields:\"whoami\") @key(fields:\"whoami\") {\n"
            + "  name: String\n"
            + "}\n"
            + "extend type PType @key(fields:\"whoami\") {\n"
            + "  somethingelse: String\n"
            + "}\n"
            ;

    @Test
    public void test() {
        SchemaParser parser = new SchemaParser();
        var typeDefinitionRegistry = parser.parse(schema);

        RuntimeWiring runtimeWiring = RuntimeWiring.newRuntimeWiring()
                .wiringFactory(new MockedWiringFactory())
                .build();

        var schemaGenerator = new SchemaGenerator();
        schemaGenerator.makeExecutableSchema(typeDefinitionRegistry, runtimeWiring);
    }
}
```

**Comment by dugenkui03** [comment]
This bug is fixed in #2147.

**Comment by robertpanzer** [comment]
Wow, that's fast! 🚀 
Thank you!
